addons/pos_self_order/controllers/orders.py

```diff
@@ -151,6 +153,19 @@ def update_existing_order(self, order, access_token, table_identifier):
         pos_order.send_table_count_notification(pos_order.table_id)
         return pos_order._export_for_self_order()
 
+    @http.route('/pos-self-order/remove-order', auth='public', type='json', website=True)
+    def remove_order(self, access_token, order_id, order_access_token):
+        pos_config = self._verify_pos_config(access_token)
+        pos_order = pos_config.env['pos.order'].browse(order_id)
+
+        if not consteq(pos_order.access_token, order_access_token) or not pos_order.exists():
```

Comment:
I think it is better to check the existence of the `pos_order` before accessing one of its field
```suggestion
        if not pos_order.exists() or not consteq(pos_order.access_token, order_access_token):
```
If you don't check its existence before and try to access a non-existant record, you'll also have a MissingError thrown but not the one you created.